src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaReviewSchedulePersistenceAdapter.kt

```diff
@@ -80,9 +83,10 @@ class JpaReviewSchedulePersistenceAdapter(
       updatedAtPrison = updateReviewScheduleStatusDto.prisonId
     }
 
-    val savedReviewScheduleEntity = reviewScheduleRepository.save(reviewScheduleEntity)
-    saveReviewScheduleHistory(savedReviewScheduleEntity)
-    return reviewScheduleEntityMapper.fromEntityToDomain(savedReviewScheduleEntity)
+    return reviewScheduleRepository.saveAndFlush(reviewScheduleEntity).let {
```

Comment:
And I think this is the other problem line, in that we were creating the history record from the saved (but not flushed) review schedule record